src/position/user-position.ts

```diff
@@ -549,6 +549,10 @@ export class UserPosition<T extends UnderlyingCollateralToken> extends PositionW
     slippage: Decimal,
     options: LeveragePositionOptions<SupportedCollateralTokens[T]> & LeveragePositionStepsPrefetch = {},
   ): AsyncGenerator<LeveragePositionStep, void, ERC20PermitSignatureStruct | undefined> {
+    if (!this.user.provider) {
+      return;
```

Comment:
instead of `return` we should throw `Error` and ask consumer to handle or else consumer will get stuck there becoz it's waiting for next step but wait forever